### PR TITLE
Implement Send-SbMessagesInBatch

### PR DESCRIPTION
**Pull request by Splaxi**
I believe this would make a great addition to the current module 😉

### CODE REVIEW

**Comment by tmeadon** [comment]
This is all looking nice!  I wonder though, playing a bit of devil's advocate here, would it be possible to just add this functionality to the existing Send-SbMessage cmdlet?  We could have two parameter sets (one for sending a single message and another for batching) and then call the relevant SbSender method depending on the invoked parameter set, a bit like what was done in Receive-SbMessage?  

I think keeping the number of exported cmdlets down where possible keeps things simple...

**Comment by Splaxi** [comment]
I'm split with this actually. I don't count the number of cmdlets / functions.

I care about that the cmdlet does one thing and one thing only. 

When working with single message vs. array of messages, abstracting that with parametersets, can increase the complexity of troubleshooting from an end user perspective.

We are using an overloaded method on the same object, but we are changing behavior and exception patterns along the way. 

When sending a single message, either the message gets inserted or not. Working with batches that picture is everything between the first message and the second last that will be inserted. 

If I need the batch capabilities, I need to understand that the exception handling is more delicate and therefore I need to take that into consideration. 

That is why I did go for a separate cmdlet. 

But it is your call entirely, I'm just trying to extend the module to fit my needs 😉

**Comment by tmeadon** [comment]
Yeah I think you're right, like you said there are more implications when working with batched messages and as a user I'd like it to be as clear as possible if batches are in use or not.  Good call!

**Comment by Splaxi** [comment]
Great! 👍

File: tests/integration/Send-SbMessagesInBatch.tests.ps1
Comment (tmeadon) [inline tests/integration/Send-SbMessagesInBatch.tests.ps1]: We just need to change the name of this test to something like "should send the correct number of messages to the correct topic"

File: tests/integration/Send-SbMessagesInBatch.tests.ps1
Comment (tmeadon) [inline tests/integration/Send-SbMessagesInBatch.tests.ps1]: Same here...

**Review by tmeadon — APPROVED** [review]
Great stuff @Splaxi - couple of very minor changes then this is good to go I reckon.

**Comment by tmeadon** [comment]
Ready to merge once that conflict is resolved!

**Pull request merged**
